src/components/Card.js

```diff
@@ -35,27 +56,59 @@ export default class Card {
     this._cardCaption.textContent = this._title;
   };
 
-  generateCard() {
+  _setLikeStatus = () => {
+    if (this._isLiked) {
+      this.likeCard();
+    } else {
+      this.dislikeCard();
+    }
+  };
+
+  _toggleLike = () => {
+    this._cardLikeButton.classList.toggle("card__heart-button_liked");
+    this._isLiked = !this._isLiked;
+    this._handleLike({ isLiked: this._isLiked, _id: this._id });
+  };
+
+  likeCard = () => {
+    this._cardLikeButton.classList.add("card__heart-button_liked");
+    this._isLiked = true;
+  };
+
+  dislikeCard = () => {
+    this._cardLikeButton.classList.remove("card__heart-button_liked");
+    this._isLiked = false;
+  };
+
+  _previewImage = () => {
+    this._handlePreviewImage(this._title, this._link);
+  };
+
+  generateCard = () => {
     this._element = this._cloneTemplate();
     this._cardImageElement = this._element.querySelector(".card__image");
     this._cardCaption = this._element.querySelector(".card__caption");
     this._cardDeleteElement = this._element.querySelector(
       ".card__trash-button"
     );
-    this._cardHeartButton = this._element.querySelector(".card__heart-button");
+    this._cardLikeButton = this._element.querySelector(".card__heart-button");
```

Comment:
Good job that you've found these variables only once here and defined them as **class fields**